### PR TITLE
Update model inference

### PR DESCRIPTION
**Pull request by adamgayoso**
This will allow us to wrap all existing methods, as well as velovi or any other new methods

- [x] Add steady-state model
- [x] Add stochastic extension
- [x] Add EM model
- [x] Add veloVI

With this paradigm we can remove `velocity` and `recover_dynamics`, and then eventually `latent_time`, etc.

We should actually make baseinference have save/load, and do it like scvi-tools:

- load is a class method
- save saves the initializing params of the instance and pickles the state dict

### CODE REVIEW

**Comment by WeilerP** [comment]
The unit tests pass locally. I ran into the linting issue in the CI before which was fixed by switch to `ruff`. So we should merge #1013 first, and then update this PR accordingly.

File: scvelo/core/_inference.py
Comment (WeilerP) [inline scvelo/core/_inference.py:1]: I originally intended to only include in `core/` only code relevant for the entire code base. What do you think? As an alternative to having `BaseInference` defined in `core/`, we could move it to e.g. `tools/_core.py`.

File: scvelo/tools/_vi_model.py
Comment (WeilerP) [inline scvelo/tools/_vi_model.py:1]: I moved the entire veloVI code into a single file `_vi_model.py`. Not super happy but not sure I like splitting it into e.g. `_vi_module.py` and `_vi_model.py` better. What do you think, @adamgayoso?

File: scvelo/tools/_steady_state_model.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    def __init__(
+        self,
+        adata: AnnData,
+        spliced_layer: str = "Ms",
+        unspliced_layer: str = "Mu",
+        constrain_ratio: Optional[Union[float, Tuple[float, float]]] = None,
+        min_r2: float = 0.01,
+        min_ratio: float = 0.01,
+        use_highly_variable: bool = True,
+        r2_adjusted: bool = True,
```
[the remark this answers is not in the document]
Comment (adamgayoso) [reply scvelo/tools/_steady_state_model.py:78]: probably an error on my part

File: scvelo/core/_inference.py
Comment (adamgayoso) [reply scvelo/core/_inference.py:1]: either way is fine with me

File: scvelo/tools/_vi_model.py
Comment (adamgayoso) [reply scvelo/tools/_vi_model.py:1]: separate files makes sense to me

Comment (WeilerP) [reply scvelo/tools/_vi_model.py:1]: Done.

File: scvelo/core/_inference.py
Comment (WeilerP) [reply scvelo/core/_inference.py:1]: Okay, I moved the constants and `BaseInference` to `tools/_core.py`.

**Review by WeilerP — DISMISSED** [review]
@adamgayoso, I'd go ahead and merge this PR for now and handle velocity, latent time, etc. in separate, smaller PRs to keep things tractable and not inflate this one. What do you think?

**Comment by WeilerP** [comment]
> That makes sense to me, but how is the EM model currently working? It doesn't implement some of the abstract methods, it's also not using the EMparams state dict. We can fix all of this in future PRs of course.

The EM model is working equivalent to `recover_dynamics`. I updated the corresponding unit tests. I'll update the class to use `EMParams`, and implement the abstract methods in a separate PR. There are also changes that need to be implemented for the steady-state model. I added corresponding TODOs.

**Pull request merged**